modules/carto/src/api/maps-v3-client.js

```diff
@@ -168,5 +192,10 @@ export async function getData({type, source, connection, credentials, geoColumn,
 
   const {accessToken} = localCreds;
 
-  return await request({url, format: mapFormat, accessToken});
+  const data = await request({url, format: mapFormat, accessToken});
+  if (mapFormat === FORMATS.CSV) {
+    return csvToGeoJson(data, {geoColumn});
```

Comment:
It's a little bit strange that getData returns the data in geoJson format when you asked for CSV.  Have in mind that this function is exposed publicly: https://deck.gl/docs/api-reference/carto/overview#support-for-other-deckgl-layers.

Reply:
I have done this now, but I changed the signature of `getData` to also return the format. Is this OK? Without also explictly returning the format I don't how we can cleanly do the conversion outside the `getData` function as the format is determined by what the map instantiation request returns